SpaceWarpMod/Utils/Expression.cs

```diff
@@ -0,0 +1,54 @@
+﻿using KontrolSystem.TO2.AST;
+using KontrolSystem.TO2.Parser;
+using KontrolSystem.TO2.Runtime;
+using KontrolSystem.TO2;
+using System.Linq;
+using KontrolSystem.Parsing;
+using KontrolSystem.KSP.Runtime.KSPConsole;
+using KontrolSystem.SpaceWarpMod.Core;
+
+namespace KontrolSystem.SpaceWarpMod.Utils {
+
+    public static class Expression {
+
+        public static object Run(string expression, KSPConsoleBuffer consoleBuffer) {
+            var result = TO2ParserREPL.REPLItems.Parse(expression);
+
+            var kspContext = new KSPContext(Mainframe.Instance.Game, consoleBuffer);
+            var registry = Mainframe.Instance.LastRegistry;
+            var context = new REPLContext(registry, kspContext);
+
+            var pollCount = 0;
+
+            try {
+                ContextHolder.CurrentContext.Value = kspContext;
+
+                foreach (var item in result.Where(i => !(i is IBlockItem))) {
+                    var future = item.Eval(context);
+                    var futureResult = future.PollValue();
+
+                    while (!futureResult.IsReady) {
+                        pollCount++;
+                        if (pollCount > 100) throw new System.Exception("No result after 100 tries");
```

Comment:
I guess we can start with this, as it will prevent commands to block the `OnGUI` loop for any significant amount of time. Though it might terminate things a bit too prematurely.